sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1521,6 +2160,99 @@ submitted in its entirety for execution via
   ...
 ----
 
+=== Dynamic Parameter Update
+
+Example showing a graph with two kernel nodes, the first created using a kernel
+bundle with `handler::set_args()`, having its node arguments updated.
+
+[source,c++]
+----
+...
+
+using namespace sycl;
+namespace sycl_ext = sycl::ext::oneapi::experimental;
+
+queue myQueue;
+auto myContext = myQueue.get_context();
+auto myDevice = myQueue.get_device();
+
+// USM allocations for kernel input/output
+const size_t n = 1024;
+int *ptrX = malloc_shared<int>(n, myQueue);
+int *ptrY = malloc_device<int>(n, myQueue);
+
+int *ptrZ = malloc_shared<int>(n, myQueue);
+int *ptrQ = malloc_device<int>(n, myQueue);
+
+// Initialize USM allocations
+
+// Kernel loaded from kernel bundle
+const std::vector<kernel_id> builtinKernelIds =
+      myDevice.get_info<info::device::built_in_kernel_ids>();
+kernel_bundle<bundle_state::executable> myBundle =
+      get_kernel_bundle(myContext, { myDevice }, builtinKernelIds);
+kernel builtinKernel = myBundle.get_kernel(builtinKernelIds[0]);
+
+// Graph containing a two kernels node
+sycl_ext::command_graph myGraph(myContext, myDevice);
+
+// Create graph dynamic parameters
+dynamic_parameter<sycl_ext::dynamic_parameter_type::usm> dynParamInput;
+dynamic_parameter<sycl_ext::dynamic_parameter_type::usm> dynParamOutput;
+dynamic_parameter<sycl_ext::dynamic_parameter_type::scalar> dynParamScalar;
+
+// First node uses ptrX as an input & output parameter, with operand
+// mySclar as another argument.
+int myScalar = 42;
+node nodeA = myGraph.add[&](handler& cgh) {
+    cgh.set_args(ptrX, myScalar);
+    cgh.parallel_for(range {n}, builtinKernel);
+});
+
+// Register nodeA dynamic parameters
+nodeA.register_dynamic_parameter(0, dynParamInput);  // Argument index 0 is ptrX
+nodeA.register_dynamic_parameter(1, dynParamScalar); // Argument index 1 is myScalar
+
+// Second node uses ptrX as an inputs, and ptrY as input/output/
+node nodeB = myGraph.add[&](handler& cgh) {
+    cgh.parallel_for(sycl::range<1>{n}, [=](sycl::id<1> it) {
+      const size_t i = it[0];
+      ptrY[i] += ptrX[i];
+    });
+   },
+   {sycl_ext::property::node::depends_on(nodeA)});
+
+// Register nodeB dynamic parameters
+nodeB.register_dynamic_parameter(ptrX, dynParamInput);
+nodeB.register_dynamic_parameter(ptrY, dynParamOutput);
+
+// Create an executable graph with the updatable property.
+auto execGraph = myGraph.finalize({sycl_ext::property::graph::updatable});
+
+// Execute graph, then update without needing to wait for it to complete
+myQueue.ext_oneapi_graph(execGraph);
+
+// Change ptrX argument to node A & B to ptrZ
+execGraph.update_dynamic_parameter(ptrZ, dynParamInput);
```

Comment:
will the below code also work?

nodeA.register_dynamic_parameter(0, dynParamInput1);
nodeB.register_dynamic_parameter(ptrX, dynParamInput2);

execGraph.update_dynamic_parameter(ptrZ, dynParamInput1);
execGraph.update_dynamic_parameter(ptrZ, dynParamInput2);

Reply:
Yup that should work